### PR TITLE
Geometric sum evaluates correctly for float base(Fix for : #11642)

### PR DESCRIPTION
**Pull request by ArifAhmed1995**
Reference : #11642 

Now geometric sum evaluates correctly for both Rational as well as float bases
Example : 

```
>>Sum((0.35)**(n), (n, 1, oo)).doit()
0.538461538461538
>>Sum((1.35)**(n), (n, 1, oo)).doit()
oo
>>Sum((0.5)**(n), (n, 1, oo)).doit()
1.00000000000000
```

Some unit tests have also been added.

### CODE REVIEW

File: sympy/concrete/tests/test_sums_products.py [hunk 1]
```diff
@@ -255,6 +255,12 @@ def test_geometric_sums():
     #issue 9908:
     assert Sum(1/(n**3 - 1), (n, -oo, -2)).doit() == summation(1/(n**3 - 1), (n, -oo, -2))
 
+    #issue 11642:
+    assert Sum(0.5**n,(n, 1, oo)).doit() == 1.0
```
Comment (smichr) [inline sympy/concrete/tests/test_sums_products.py]: I suspect this comes back as `1`, not `1.0`. I think it should come back as a Float: if the input is not exact, the output should not be exact. You can test the result by `assert result.is_Float` in addition to testing the value (`result == 1`).

File: sympy/matrices/expressions/matexpr.py [hunk 2]
```diff
@@ -127,12 +127,20 @@ def __pow__(self, other):
             raise ShapeError("Power of non-square matrix %s" % self)
         elif self.is_Identity:
             return self
-        elif other is S.NegativeOne:
+        elif other is S.NegativeOne and not isinstance(self,MatPow):
```
Comment (smichr) [inline sympy/matrices/expressions/matexpr.py]: space after commas here and below

File: sympy/matrices/expressions/tests/test_matpow.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def test_as_explicit():
     assert MatPow(A, -2).as_explicit() == (A.inv())**2
     # less expensive than testing on a 2x2
     A = ImmutableMatrix([4]);
-    assert MatPow(A, S.Half).as_explicit() == A**S.Half
+    assert MatPow(A, S.Half).as_explicit() == (A**S.Half ).doit()
```
Comment (smichr) [inline sympy/matrices/expressions/tests/test_matpow.py]: no space before `)`

File: sympy/matrices/matrices.py [hunk 4]
```diff
@@ -208,16 +207,15 @@ def __pow__(self, num):
...
             def jordan_cell_power(jc, n):
                 N = jc.shape[0]
                 l = jc[0, 0]
                 for i in range(N):
-                        for j in range(N-i):
-                                bn = binomial(n, i)
-                                if isinstance(bn, binomial):
-                                        bn = bn._eval_expand_func()
-                                jc[j, i+j] = l**(n-i)*bn
+                    for j in range(N-i):
```
Comment (smichr) [inline sympy/matrices/matrices.py]: space around `-`

File: sympy/matrices/matrices.py [hunk 5]
```diff
@@ -226,8 +224,20 @@ def jordan_cell_power(jc, n):
                 jordan_cell_power(j, num)
             return self._new(P*diag(*jordan_cells)*P.inv())
         else:
-            raise TypeError(
-                "Only SymPy expressions or int objects are supported as exponent for matrices")
+            raise TypeError("Only SymPy expressions or int objects are supported as exponent for matrices")
```
Comment (smichr) [inline sympy/matrices/matrices.py]: "supported as matrix exponents."

File: sympy/matrices/tests/test_matrices.py [hunk 6]
```diff
@@ -196,24 +196,29 @@ def test_power():
     assert Matrix([[1, 2], [3, 4]])**Integer(2) == Matrix([[7, 10], [15, 22]])
 
     A = Matrix([[33, 24], [48, 57]])
-    assert (A**(S(1)/2))[:] == [5, 2, 4, 7]
+    assert ( (A**(S(1)/2)).doit() )[:] == [5, 2, 4, 7]
```
Comment (smichr) [inline sympy/matrices/tests/test_matrices.py]: remove space after/before `)`

**Review by smichr — CHANGES_REQUESTED** [review]

**Comment by ArifAhmed1995** [comment]
@smichr The commit history is terribly messed up as two series of commits from separate fixes got intermingled.I've deleted the branch and closed the PR.

I have implemented the changes requested in two separate PRs : #11682 and #11648 . Can you look through the changes now?

**Pull request closed without merging**